File: SSPMessageFactory.java
package burp;

import java.util.Arrays;

public class SSPMessageFactory {

    public static SSPMessage getMessage(byte[]  SSPHeader) {
        byte[] versionMessage = Arrays.copyOfRange(SSPHeader, 8, 12);

        int version = new Byte(versionMessage[0]).intValue();

        switch (version) {
            case 1:
                return new SSPRequest(SSPHeader);
            case 2:
                return new SSPChallenge(SSPHeader);
            case 3:
                return new SSPResponse(SSPHeader);
            default:
                return null;
        }
    }

}

File: src/main/kotlin/burp/messages/SSPMessageFactory.kt
package burp.messages

import java.util.*

object SSPMessageFactory {

    fun getMessage(sspHeader: ByteArray): SSPMessage {
        val versionMessage = Arrays.copyOfRange(sspHeader, 8, 12)

        val version = versionMessage[0].toInt()

        when (version) {
            1 -> return SSPRequest(sspHeader)
            2 -> return SSPChallenge(sspHeader)
            3 -> return SSPResponse(sspHeader)
            else -> return SSPEmpty(sspHeader)
        }
    }

}